File: FolkLibrary.Shared/Dtos/AlbumReadDto.cs
namespace FolkLibrary.Dtos;

public sealed class AlbumReadDto : AlbumReadDtoBase
{
    public HashSet<ArtistReadDtoBase> Artists { get; set; } = null!;
    public List<TrackReadDtoBase> Tracks { get; set; } = null!;
}

File: FolkLibrary/ClientApp/src/app/models/album-read-dto.ts
import { AlbumReadDtoBase } from "./album-read-dto-base";
import { ArtistReadDtoBase } from "./artists-read-dto-base";
import { ItemReadDto } from "./item-read-dto";
import { TrackReadDtoBase } from "./track-read-dto-base";

export interface AlbumReadDto extends AlbumReadDtoBase {
    readonly artists: ArtistReadDtoBase[];
    readonly genres: ItemReadDto[];
    readonly tracks: TrackReadDtoBase[];
}